keps/sig-multicluster/2149-clusterid/README.md

```diff
@@ -474,14 +678,40 @@ spec:
 
 #### `clusterset.k8s.io ClusterProperty`
 
-A cluster in a ClusterSet is expected to be authoritatively associated with that ClusterSet by an external process and storage mechanism with a purview above the cluster local boundary, whether that is some form of a cluster registry or just a human running kubectl. (The details of any specific mechanism is out of scope for the MCS API and this KEP -- see the Non-Goals section.) Mirroring this information in the cluster-local `ClusterProperty` CRD will necessarily need to be managed above the level of the cluster itself, since the properties of `clusterset.k8s.io` extend beyond the boundaries of a single cluster, and will likely be something that has access to whatever cluster registry-esque concept is implemented for that multicluster setup. It is expected that the mcs-controller ([as described in the MCS API KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api#proposal)), will act as an admission controller to verify individual objects of this property.
-
-Because there are obligations of the `cluster.clusterset.k8s.io ClusterProperty` that are not meanigfully verifiable until a cluster tries to join a ClusterSet and set its `clusterset.k8s.io ClusterProperty`, the admission controller responsible for setting a `clusterset.k8s.io ClusterProperty` will need the ability to reject such an attempt when it is invalid, and alert `[UNRESOLVED]` or possibly affect changes to that cluster's `cluster.clusterset.k8s.io ClusterProperty` to make it valid `[/UNRESOLVED]`. Two symptomatic cases of this would be:
-
-1. When a cluster with a given `cluster.clusterset.k8s.io ClusterProperty` tries to join a ClusterSet, but a cluster with that same `cluster.clusterset.k8s.io ClusterProperty` appears to already be in the set.
-2. When a cluster that does not have a `cluster.clusterset.k8s.io ClusterProperty` tries to join a ClusterSet.
-
-In situations like these, the admission controller will need to fail to add the invalid cluster to the ClusterSet by refusing to set its `clusterset.k8s.io ClusterProperty`, and surface an error that is actionable to make the property valid.
+A cluster in a ClusterSet is expected to be authoritatively associated with that
```

Comment:
I think I would soften all of this to make less assumptions.  A self-assembling (e.g. peer-to-peer discovered) clusterset should be a valid implementation